### PR TITLE
Automapper part I: triangulation pairs grid

### LINKED ISSUE #38: Ability to save and load triangulation point pairs
https://discord.com/channels/612683871112396800/612685195144855582/1059976142997700650

### PR DESCRIPTION
**Pull request by IhateTrains**
This is the first major step towards autogenerating province mappings from triangulation point pairs. Splitting work into at least two pull requests because it's already quite a lot to review.

- The current PR doesn't implement the province mapping autogeneration. It only adds the reading, editing and saving of triangulation pairs.
- I haven't removed or modified the existing 3-pair triangulation feature yet. It may become obsolete when the persistent triangulation points are actually used for triangulation, which is not the case yet.
- The points are drawn on the canvas like the ones from the existing triangulation feature, except they are grey when inactive and blue for the currently selected pair.
- The pairs are displayed in a grid separate from the province mappings grid.
- The pairs can be added with the F6 key and removed with the Del key.

Menu item for adding a new triangulation point pair:
obraz

Scrollable triangulation pairs grid above the province mappings grid:
obraz

Example triangulation pairs for Ireland. The active one is blue, the rest is grey:
obraz

Snippet of the generated province mappings file:
```
imperator_invictus = {
	triangulation_pair = { srcX = 424 srcY = 761 dstX = 434 dstY = 1267 }
	triangulation_pair = { srcX = 403 srcY = 697 dstX = 408 dstY = 1217 }
	triangulation_pair = { srcX = 504 srcY = 616 dstX = 490 dstY = 1136 }
	triangulation_pair = { srcX = 653 srcY = 639 dstX = 634 dstY = 1142 }
	triangulation_pair = { srcX = 624 srcY = 739 dstX = 615 dstY = 1233 }
	triangulation_pair = { srcX = 698 srcY = 550 dstX = 670 dstY = 1061 }
	triangulation_pair = { srcX = 624 srcY = 449 dstX = 589 dstY = 980 }
	triangulation_pair = { srcX = 457 srcY = 526 dstX = 436 dstY = 1067 }
	link = { comment = "=== NEW MAPPINGS BY IHT ===" }
	link = { comment = "ETHIOPIA BY IHT" }
	link = { imp = 7552 imp = 7553 ck3 = 6380 } # Deresa, Aqiq -> BADI
```

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Zemurin) [inline .idea/workspace.xml]: This file shouldn't be necessary, please put it in gitignore.

**Review by Zemurin — APPROVED** [review]
I get what you're doing, approximately, but I still have no idea how you'll manage automapping within a triangulation mesh. Looking forward to seeing it! :)

File: .idea/workspace.xml (hunk 1 above)
Comment (IhateTrains) [reply .idea/workspace.xml]: Somehow missed it when reviewing the changes, fixed

**Comment by IhateTrains** [comment]
> I get what you're doing, approximately, but I still have no idea how you'll manage automapping within a triangulation mesh. Looking forward to seeing it! :)

From the research I did earlier, the Delaunay triangulation algorithm should help. Here's a fragment of what MS Copilot replied to my prompt:

> To find the smallest triangle containing a given point (XY) from a set of points (A, B, C, D, E, etc.), you can use a computational geometry algorithm known as the Delaunay triangulation.
> Here is a high-level description of how you might implement this:
> - Compute the Delaunay triangulation for your set of points. The Delaunay triangulation is a triangulation such that no point is inside the circumcircle of any triangle. It maximizes the minimum angle of all the angles of the triangles in the triangulation.
> - Find the triangle that contains the point XY. You can do this by checking each triangle from the Delaunay triangulation to see if it contains the point XY. A point P is inside a triangle ABC if the areas of ABC, PBC, PCA, and PAB are all positive.
> - Return the vertices of the smallest triangle. If there are multiple triangles that contain the point XY, return the one with the smallest area.

**Pull request merged**